rust/automerge/src/transaction/inner.rs

```diff
@@ -263,6 +302,13 @@ impl TransactionInner {
         value: V,
     ) -> Result<(), AutomergeError> {
         let obj = doc.exid_to_obj(ex_obj)?;
+        let obj_type = doc
```

Comment:
This `let obj_type = ...` stanza appears a lot. Worth adding a little helper function in this file like `obj_type` which returns an `AutomergeError` directly so we don't have to keep doing the `ok_or` bit?